subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/AbiExtractingClasspathResourceHasher.java

```diff
@@ -70,8 +70,13 @@ public HashCode hash(ZipEntry zipEntry) throws IOException {
         if (!isClassFile(zipEntry.getName())) {
             return null;
         }
-        byte[] content = new byte[zipEntry.size()];
-        ByteStreams.readFully(zipEntry.getInputStream(), content);
+        byte[] content;
+        if (zipEntry.size() >= 0) {
```

Comment:
that's a possibility? I.e. they're not storing the size? How did you find out?

Reply:
You don't get the size when you have a ZipInputStream. The size of each entry is only listed at the end of a zip file.